tests/test_requests.py

```diff
@@ -221,6 +221,23 @@ def test_http_303_doesnt_change_head_to_get(self, httpbin):
         assert r.history[0].status_code == 303
         assert r.history[0].is_redirect
 
+    def test_header_and_body_removal_on_redirect(self, httpbin):
+        purged_headers = ('Content-Length', 'Content-Type', 'Transfer-Encoding')
```

Comment:
This test doesn't actually cover 'Transfer-Encoding' since 'Content-Length' is present. I can add a smaller separate test for 'Transfer-Encoding' if it's deemed that we definitely want to remove it.

This does ensure that as long as `purged_headers` contains 'Transfer-Encoding' in `resolve_redirects`, that it will be removed if it occurs. It just won't fail for accidental removal from the `purged_headers` tuple.